aya/src/bpf.rs

```diff
@@ -580,12 +605,22 @@ impl<'a> BpfLoader<'a> {
                 (name, program)
             })
             .collect();
-        let maps: Result<HashMap<String, Map>, BpfError> = maps.drain().map(parse_map).collect();
+        let maps = maps
+            .drain()
+            .map(parse_map)
+            .collect::<Result<HashMap<String, Map>, BpfError>>()?;
+
+        if !self.allow_unsupported_maps
+            && maps
+                .iter()
+                .filter(|(_, x)| matches!(x, Map::Unsupported(_)))
+                .count()
```

Comment:
This should use `any`

Reply:
So I ended up changing this logic a bit in order to support `MapError::Unsupported { map_type: u32 }`

More specifically we only search our maps if allow_unsupported_maps is set.